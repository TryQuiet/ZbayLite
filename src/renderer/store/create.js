import { applyMiddleware, createStore, compose } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const composer = (enhancers) => {
  if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      process.env.NODE_ENV === 'development') {
    return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__(enhancers)
  }
  return compose(enhancers)
}

export default ({
  initialState
} = {}) => createStore(
  reducers,
  initialState,
  composer(applyMiddleware(...[errorsMiddleware, createDebounce(), thunk, promise()]))
)
