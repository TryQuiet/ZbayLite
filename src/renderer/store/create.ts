import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'
import createSagaMiddleware from 'redux-saga'
import rootSaga from '../sagas/index.saga'
import { composeWithDevTools } from 'redux-devtools-extension'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const sagaMiddleware = createSagaMiddleware()

export default (initialState = {}) => {
  const store = createStore(
    reducers,
    initialState,
    composeWithDevTools(
      applyMiddleware(...[errorsMiddleware, createDebounce(), sagaMiddleware, thunk, promise()])
    )
  )
  sagaMiddleware.run(rootSaga)
  return store
}
