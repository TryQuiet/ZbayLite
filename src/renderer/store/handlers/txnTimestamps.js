import { produce } from 'immer'
import { handleActions, createAction } from 'redux-actions'

import { actionTypes } from '../../../shared/static'

export const initialState = {}
export const addTxnTimestamp = createAction(actionTypes.ADD_TXN_TIMESTAMP)
export const actions = {
  addTxnTimestamp
}
export const getTnxTimestamps = () => async (dispatch, getState) => {}
export const epics = {
  getTnxTimestamps
}

export const reducer = handleActions(
  {
    [addTxnTimestamp]: (state, { payload: { tnxs } }) => produce(state, (draft) => {
      return {
        ...draft,
        ...tnxs
      }
    })
  },
  initialState
)

export default {
  reducer,
  actions,
  epics
}
