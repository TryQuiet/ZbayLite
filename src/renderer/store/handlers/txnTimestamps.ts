import { produce } from 'immer'
import { handleActions, createAction } from 'redux-actions'

import { ActionsType, PayloadType } from './types'

import { actionTypes } from '../../../shared/static'

interface ITxnTimestamps {
  [id: number]: string
}

export type TxnTimestampsStore = ITxnTimestamps

export const initialState: ITxnTimestamps = {}
export const addTxnTimestamp = createAction<{
  tnxs: {
    [id: number]: string
  }
}>(actionTypes.ADD_TXN_TIMESTAMP)

export const actions = {
  addTxnTimestamp
}

export type TxnTimestampsActions = ActionsType<typeof actions>

export const reducer = handleActions<TxnTimestampsStore, PayloadType<TxnTimestampsActions>>(
  {
    [addTxnTimestamp.toString()]: (state, { payload: { tnxs } }: TxnTimestampsActions['addTxnTimestamp']) => produce(state, (draft) => {
      return {
        ...draft,
        ...tnxs
      }
    })
  },
  initialState
)

export default {
  reducer,
  actions,
}
