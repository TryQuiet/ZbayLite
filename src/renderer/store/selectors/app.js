import { createSelector } from 'reselect'

const store = s => s

export const app = createSelector(store, state => state.get('app'))

const version = createSelector(app, a => a.version)
const useTor = createSelector(app, a => a.useTor)
const transfers = createSelector(app, a => a.transfers)
const currentModalTab = createSelector(app, a => a.modalTabToOpen)
const allTransfersCount = createSelector(app, a => a.allTransfersCount)
const newTransfersCounter = createSelector(app, a => a.newTransfersCounter)
const isInitialLoadFinished = createSelector(app, a => a.isInitialLoadFinished)
const directMessageQueueLock = createSelector(
  app,
  a => a.directMessageQueueLock
)
const messageQueueLock = createSelector(app, a => a.messageQueueLock)

export default {
  version,
  transfers,
  currentModalTab,
  allTransfersCount,
  newTransfersCounter,
  messageQueueLock,
  directMessageQueueLock,
  isInitialLoadFinished,
  useTor
}
