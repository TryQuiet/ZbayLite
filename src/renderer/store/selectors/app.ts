import { createSelector } from 'reselect'

import { Store } from '../reducers'

const app = (s: Store) => s.app

const version = createSelector(app, (a) => a.version)
const currentModalTab = createSelector(app, (a) => a.modalTabToOpen)
const isInitialLoadFinished = createSelector(
  app,
  (a) => a.isInitialLoadFinished
)

export default {
  version,
  currentModalTab,
  isInitialLoadFinished
}
