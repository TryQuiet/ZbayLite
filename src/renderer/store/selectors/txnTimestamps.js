import { createSelector } from 'reselect'
const store = s => s

const tnxTimestamps = createSelector(
  store,
  state => state.txnTimestamps
)

export default {
  tnxTimestamps
}
