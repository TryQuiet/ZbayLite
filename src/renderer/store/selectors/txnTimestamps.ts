import { TxnTimestampsStore } from '../handlers/txnTimestamps'

const tnxTimestamps = (s): TxnTimestampsStore => s.txnTimestamps as TxnTimestampsStore

export default {
  tnxTimestamps
}
