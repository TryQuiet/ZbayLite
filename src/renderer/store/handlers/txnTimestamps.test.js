import create from '../create'
import { actions, initialState, epics } from './txnTimestamps'
import selectors from '../selectors/txnTimestamps'

describe('Operations reducer handles ', () => {
  let store = null

  beforeEach(() => {
    store = create({
      initialState: {
        txnTimestamps: {
          ...initialState
        }
      }
    })
    jest.clearAllMocks()
  })

  describe('actions', () => {
    it('- addTxnTimestamp', () => {
      store.dispatch(actions.addTxnTimestamp({ tnxs: { 123: '1234' } }))
      expect(selectors.tnxTimestamps(store.getState())).toMatchSnapshot()
    })
  })
  describe('epics', () => {
    it('- getTnxTimestamps', async () => {
      await store.dispatch(epics.getTnxTimestamps())
      expect(selectors.tnxTimestamps(store.getState())).toMatchSnapshot()
    })
  })
})
