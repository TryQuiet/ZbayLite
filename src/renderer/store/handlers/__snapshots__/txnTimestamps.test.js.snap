// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operations reducer handles  actions - addTxnTimestamp 1`] = `
Object {
  "123": "1234",
}
`;

exports[`Operations reducer handles  epics - getTnxTimestamps 1`] = `Object {}`;
